gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/QueryStage.java

```diff
@@ -373,8 +373,9 @@ public String toString() {
 
   @Override
   public int hashCode() {
```

Comment:
I think there should be 2 options for testing (other than using the service as is via a local or remote helper):

1) Don't mock use as is (and class is final). Typically for data types that a user can fully configure for its tests.
2) This class can't (or shouldn't) be created by the user and therefore have to be mocked for tests.
  In this case, class should not be final, constructor should be typically package scope and hashCode/equals methods should be final.

As this class is not final (and I didn't take the time to see why - other than seeing if it has subclasses) I assumed
it is part of (2).